packages/integration-sdk-runtime/src/logger/registerEventEmitters.ts

```diff
@@ -73,25 +91,27 @@ export function unregisterEventEmitters({
   uncaughtExceptionListener,
 }: RegisteredEventEmitters) {
   process.nextTick(() => {
-    process.removeListener(
-      'multipleResolves',
-      multipleResolveListener,
-    );
-    process.removeListener(
-      'unhandledRejection',
-      unhandledRejectionListener,
-    );
-    process.removeListener(
-      'uncaughtException',
-      uncaughtExceptionListener,
-    );
+    process.removeListener('multipleResolves', multipleResolveListener);
+    process.removeListener('unhandledRejection', unhandledRejectionListener);
+    process.removeListener('uncaughtException', uncaughtExceptionListener);
   });
 }
 
+interface MinimalLogger {
+  error: IntegrationLogger['error'];
+  onFailure?: IntegrationLogger['onFailure'];
+}
+
 function integrationLoggerEventEmitterCallback(
-  getLogger: () => Pick<IntegrationLogger, 'error'>,
+  getLogger: () => MinimalLogger,
 ): LifecycleErrorCallback {
-  return (err, event) => getLogger().error({ err, event });
+  return (err, event) => {
+    const logger = getLogger();
+    logger.error({ err, event });
+    if (logger.onFailure) {
+      logger.onFailure({ err });
+    }
+  };
```

Comment:
Here's the change in this file. The rest is the addition of the `MinimalLogger` interface or changes the linter made.